### ISSUE TITLE
Question regarding net_d

### ISSUE THREAD

**Issue by seanko29** [muslll/neosr#78]
Hi muslll,

Thank you for your great work. It works perfectly fine in python 3.10 and fixed the yaml configuration.
I had another question and I guess it was better if I open a new issue thread. 

For networks such as SAFMN and SPAN, there are no net_d (no discriminator) from what I read in the paper. 
However, when. I run train_span.yml or train_safmn.yml in their basic setting equivalent with the repo, it seems to work with the discriminator turned on and training goes smoothly.

So the question is, 
are there explicit different results when we use a UNet discriminator + gan_loss on net_d on networks that does not essentially require a net_d? (like span or safmn) or does it improve the performance when we add the net_d during training for any models? 

Thank you in advanced :).

**Comment by muslll** [comment]
Hi, glad you got it working.
About GAN and discriminator: it works regardless of the network. GAN is usually applied to real-world SISR to push the network to generate more perceptually pleasant images. The default configurations on neosr of all networks targets the best real-world SISR practices, so in all of them GAN is enabled. Neosr also supports different [kinds of discriminators](https://github.com/muslll/neosr?tab=readme-ov-file#supported-discriminators) that you can try out. Unet-SN is standard, but `a2fpn` also works pretty well (sometimes even better, depending on your dataset and LQ degradations).

**Comment by seanko29** [comment]
So, if I understood your reply correctly,
Q1. discriminators in neosr boost real world SISR and does it work independently from the generator network?
Q2. If I am looking at more research-focused SR, is it perfectly fine to turn-off the discriminator setting first during my experiments? 

Thank you.

**Comment by muslll** [comment]
Yes to both Q1 and Q2. About Q2: usually when training from scratch (no pretrained network), you *should* disable GAN and train with only [`pixel_opt`](https://github.com/muslll/neosr/wiki/Losses#pixel_opt) or [`mssim`](https://github.com/muslll/neosr/wiki/Losses#mssim_opt) first. GAN is only used after you have a stable model.

**Comment by Phhofm** [comment]
Just to add to the understanding (its whats already written here, just re-stated): net_d has nothing to do with the network. 

Since you listed SPAN and SAFMN, it doesnt matter, you could list all of them here (ATD, SRFormer, HAT, SwinIR, ESRGAN, etc) or not a single one. It is more coupled to a loss, gan loss, not a network.
So none of those require a net_d. All of them can use it. Its not the network. (im just stating my points in the extreme to make it more clear)

An official pretrain almost never has a net_d because in most cases they are trained with pixel loss only on bicubic downsampled dataset like div2k or df2k. Thats what you will most likely find in papers or in official github repositories. Unless they release a real world model where they use the Real-ESRGAN degradation pipeline or something similiar, where they add noise, blur, compression to the dataset so the model can deal with those.

If you want to see a manual example of this you can have a look at my [4xNomosWebPhoto Dataset PDF](file:///home/phhofm/Downloads/4xNomosWebPhoto.pdf) where I showed how I created it, it is also attached in the release of my [4xNomosWebPhoto_realplksr](https://github.com/Phhofm/models/releases/tag/4xNomosWebPhoto_RealPLKSR) model.

Hm maybe not the best example but you could have a look at what I used for my [SPAN pretrains](https://github.com/Phhofm/models/releases/tag/4xSPAN_pretrains) since I included the config files in the attachements where I basically deactivated any loss except pixel (or mssim) and ran on a downsampled-only lr.

Maybe this helps a bit, simply wanted to add to the already answers here.
